src/binaryutils.cpp

```diff
@@ -186,13 +186,17 @@ Status validateResolutionAgainstFirstBatchImage(const cv::Mat input, cv::Mat* fi
 
 bool checkBatchSizeMismatch(const std::shared_ptr<TensorInfo>& tensorInfo,
     const int batchSize) {
-    if (batchSize <= 0)
+    auto d = tensorInfo->getBatchSize();
+    return !Dimension(d ? d : DYNAMIC_DIMENSION).match(batchSize);
```

Comment:
We should hide implementation detail - `DYNAMIC_DIMENSION=-1` in private field of `Dimension` class.  If we dont do that, we will always end up moving implementation details to places we regret afterwards.

My idea is to hide `DYNAMIC_DIMENSION` and use `Dimension::any()` here.